exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -157,7 +157,7 @@ func (m *AzureManagedMachinePool) ValidateUpdate(oldRaw runtime.Object, client c
 			allErrs = append(allErrs, field.Invalid(
```

Comment:
I wonder if `Forbidden` would be a more appropriate error type here

> // Forbidden returns a *Error indicating "forbidden".  This is used to
// report valid (as per formatting rules) values which would be accepted under
// some conditions, but which are not permitted by current conditions (e.g.
// security policy).

Reply:
I like it